sharpd/sharp_vty.c

```diff
@@ -161,7 +163,14 @@ DEFPY (install_routes,
        "sharp install routes [vrf NAME$vrf_name]\
 	  <A.B.C.D$start4|X:X::X:X$start6>\
 	  <nexthop <A.B.C.D$nexthop4|X:X::X:X$nexthop6>|\
-	   nexthop-group NHGNAME$nexthop_group>\
+	   nexthop-group NHGNAME$nexthop_group|\
+	   nexthop-seg6local NAME$seg6l_oif\
+	     <End$seg6l_end|\
+	      End_X$seg6l_endx X:X::X:X$seg6l_endx_nh6|\
+	      End_T$seg6l_endt (1-4294967295)$seg6l_endt_table|\
+	      End_DX4$seg6l_enddx4 A.B.C.D$seg6l_enddx4_nh4|\
+	      End_DT6$seg6l_enddt6 (1-4294967295)$seg6l_enddt6_table>|\
+	   nexthop-seg6$nh_seg6 X:X::X:X$seg6_nh6 encap X:X::X:X$seg6_seg>\
```

Comment:
you're adding something like a dozen new arguments here - it's a bit unwieldy: could this variation be split into its own function?

Reply:
> it's a bit unwieldy

I'm agree with your comment.

> could this variation be split into its own function?

Is your point that "these SRv6 settings should be designed as a separate command from sharp-install-routes"?

It's like sharp-install-seg6local-routes, sharp-install-seg6-routes.

Reply:
@mjstapp 

https://github.com/FRRouting/frr/pull/5865/commits/82176ff0bf91c397dc427db09ddfe17986c40780

I added (1) "sharp install seg6-route" (2) "sharp install seg6local-route".

**Why seg6/seg6local are implemented as separate commands?** 

The reason is that seg6 and seg6local are two different network processes, seg6 is SRv6 Headend and seg6local is SRv6 Endpoint.

For example, in SRv6 Endpoint method, prefix is always in IPv6 format, while in SRv6 Headend method, prefix can be in either IPv4 or IPv6 format.

So by keeping them separate commands, it is much simpler and less confusing than using a single "sharp install srv6-route" command.